spotifycli/spotifycli.py

```diff
@@ -212,45 +246,63 @@ def show_art_url():
+        raise SpotifyCLIException(
+            "Could not retrieve list of services (bus names) from dbus." +
+            f"Full error text:\n{reply.header.message}"
         )
-        connection = open_dbus_connection(bus="SESSION")
-
-        list_names_call = new_method_call(
-            remote_obj=dbus_addr, method="ListNames", signature=""
-        )
-        reply = connection.send_and_get_reply(list_names_call)
-        names = reply.body[0]
-
-        client_name = f"org.mpris.MediaPlayer2.{client}"
-        mpris_name = next((name for name in names if name.startswith(client_name)), None)
-        if mpris_name is None:
-            sys.stderr.write(f"No mpris clients found for client {client}\n")
-            sys.exit(1)
-
-        spotify_dbus_addr = DBusAddress(
-            bus_name=mpris_name,
-            object_path="/org/mpris/MediaPlayer2",
-            interface="org.freedesktop.DBus.Properties"
-        )
-        get_property_call = new_method_call(
-            remote_obj=spotify_dbus_addr,
-            method="Get",
-            signature="ss",
-            body=("org.mpris.MediaPlayer2.Player", spotify_property)
-        )
-
-        reply = connection.send_and_get_reply(get_property_call)
-        body = reply.body[0]
-        return body[1]
+    names = reply.body[0]
+
+    client_name = f"org.mpris.MediaPlayer2.{client}"
+    mpris_name = next(
+        (name for name in names if name.startswith(client_name)), None
+    )
+    if mpris_name is None:
+        raise SpotifyCLIException(
+            f"No mpris clients found for {client}, is {client} running?\n")
+
+    spotify_dbus_addr = DBusAddress(
+        bus_name=mpris_name,
+        object_path="/org/mpris/MediaPlayer2",
+        interface="org.freedesktop.DBus.Properties"
+    )
+    get_property_call = new_method_call(
+        remote_obj=spotify_dbus_addr,
+        method="Get",
+        signature="ss",
+        body=("org.mpris.MediaPlayer2.Player", spotify_property)
+    )
+    reply = connection.send_and_get_reply(get_property_call)
+    if reply.header.message_type.name == 'error':
+        err = '>  ' + ' ...\n>  '.join(textwrap.wrap(reply.body[0], 120))
+        if 'AppArmor policy prevents' in reply.body[0]:
+            info = "Could not connect to Spotify instance.\n"
+            info += "It appears AppArmor has the current app sandboxed.\n"
+            info += "This can block method calls via dbus.\n"
+            info += "Try running from a different environment,\n"
+            info += " like the default terminal.\n"
+            info += "Info about AppArmor and sandboxing can be found at:\n"
+            info += " https://ubuntu.com/core/docs/security-and-sandboxing\n"
+            info += f"Full system response:\n{err}"
+            raise SpotifyCLIException(info)
+        else:
+            raise SpotifyCLIException(
+                f"Could not connect to Spotify instance, full response:\n{err}"
+            )
```

Comment:
And here is the fix that started all this :upside_down_face: 

Previously this assumed the reply.body was valid, and resulted in a confusing exception from Python.  Now that actually checks the reply header to see if it's an error, builds a SpotifyCLIException, and lets that bubble up to the caller.  (`main()`)

In addition, this provide some diagnostic help.  If it detects the AppArmor message in the error text, it offers some guidance to the user about what might be wrong, how to work around, and how to find more info.